public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -130,22 +131,39 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
       format: 'epoch_millis',
     };
 
+    const rangeEnd: any = {};
+    if (timeEndField) {
+      rangeEnd[timeEndField] = {
+        from: options.range.from.valueOf(),
+        to: options.range.to.valueOf(),
+        format: 'epoch_millis',
+      };
+    }
+
     const queryInterpolated = this.templateSrv.replace(queryString, {}, 'lucene');
     const query = {
       bool: {
         filter: [
-          { range: range },
           {
             query_string: {
               query: queryInterpolated,
             },
           },
         ],
+        should: [{ range: range }],
+        minimum_should_match: 1,
       },
     };
+    if (rangeEnd) {
+      query.bool.should.push({ range: rangeEnd });
+    }
+
+    const sort: any = {};
+    sort[timeField] = 'asc';
 
     const data: any = {
       query,
+      sort: [sort],
```

Comment:
Why sort ascending, isn't that the default?

Reply:
First the reason why I added the sort-info in general:

I believe the default is actually to sort on score, then date - and because the old way used a non-scoring `filter`-query, the scores were always 0, and then the sort ended up being the date-field.

In order to search for annotations where either the start or end-time of the annotation-region are inside the current dasbhard-range, then the query had to be modified to a `should` bool-query with a `min_should_match: 1`-param.

Because the `should`-filter is a scoring filter, then as I understand it, annotations with both the start and end-data inside the region would score higher, and thus come out before regions with one the start or end-date inside the range.

---

And then you're actual question:

You are correct in that the default sorting is `asc`, so the code could probably be simplified into simply `sort: [ timeField ]` instead of `sort: [ {[timeField]: 'asc' }]` but I personally like the explicitness of having the sort-order in the code. 

But if you would prefer the simpler code relying on the default-sort, I'll swap it out.

Reply:
Thanks. Wouldn't be more correct to use two range-filters, one for `from` of `timeField` and one for `to` of `timeFieldEnd` if both `timeField` and `timeFieldEnd` are configured?

Reply:
Hmm, that might work, yes - I'll have to test that out. I'll probably have time to test it tomorrow.

If you a `timeField` and `timeFieldEnd` is defined, it currently builds a query like so:

```json
{
  "bool": {
    "should": [
      { "range": { "@timestamp": { "from": 0, "to": 2, "format": "epoch_millis" } } },
      { "range": { "@timestampEnd": { "from": 0, "to": 2, "format": "epoch_millis" } } }
    ],
    "minimum_should_match": 1
  }
}
```

But then it turns into a scoring query (because they are in the `should`-part of the `bool`-filter) - I hadn't considered nesting two `bool`-queries to make the `should`-query into a non-scoring query.

I don't know which query is the cleanest looking - the un-nested one with a sort, or a nested one that doesn't need a sort.
But there's a chance the nested one is faster, because it doesn't need to perform the sort, so if that works, I could be tempted into leaning towards the nested option you propose.